src/isomorphic/modern/class/__tests__/ReactClassEquivalence-test.js

```diff
@@ -11,24 +11,50 @@
 
 'use strict';
 
-var MetaMatchers = require('MetaMatchers');
+var spawnSync = require('child_process').spawnSync;
+var path = require('path');
 
 describe('ReactClassEquivalence', function() {
-
-  beforeEach(function() {
-    this.addMatchers(MetaMatchers);
-  });
-
-  var es6 = () => require('./ReactES6Class-test.js');
-  var coffee = () => require('./ReactCoffeeScriptClass-test.coffee');
-  var ts = () => require('./ReactTypeScriptClass-test.ts');
-
   it('tests the same thing for es6 classes and CoffeeScript', function() {
-    expect(coffee).toEqualSpecsIn(es6);
+    var result1 = runJest('ReactCoffeeScriptClass-test.coffee');
+    var result2 = runJest('ReactES6Class-test.js');
+    compareResults(result1, result2);
   });
 
   it('tests the same thing for es6 classes and TypeScript', function() {
-    expect(ts).toEqualSpecsIn(es6);
+    var result1 = runJest('ReactTypeScriptClass-test.ts');
+    var result2 = runJest('ReactES6Class-test.js');
+    compareResults(result1, result2);
   });
 
 });
+
+function runJest(testFile) {
+  var cwd = process.cwd();
+  var jestBin = path.resolve('node_modules', '.bin', 'jest');
+  var result = spawnSync('node', [jestBin, testFile, '--verbose'], {cwd});
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  if (result.status !== 0) {
+    throw new Error(
+      'jest process exited with: ' +
+      result.status +
+      '\n' +
+      'stdout: ' +
+      result.stdout.toString() +
+      'stderr: ' +
+      result.stderr.toString()
+    );
+  }
+
+  return result.stdout.toString();
+}
+
+function compareResults(a, b) {
+  var aSpecs = (a.match(/it\s.*$/gm) || []).sort().join('\n');
+  var bSpecs = (b.match(/it\s.*$/gm) || []).sort().join('\n');
+  expect(aSpecs).toEqual(bSpecs);
+}
```

Comment:
Note that this only checks the list of tests

```
  ✓ it preserves the name of the class for use in error messages
  ✓ it throws if no render function is defined
  ✓ it renders a simple stateless component with prop
  ✓ it renders based on state using initial values in this.props
  ✓ it renders based on state using props in the constructor
  ✓ it renders based on context in the constructor
  ✓ it renders only once when setting state in componentWillMount
  ✓ it should throw with non-object in the initial state property
  ✓ it should render with null in the initial state property
  ✓ it setState through an event handler
  ✓ it should not implicitly bind event handlers
  ✓ it renders using forceUpdate even when there is no state
  ✓ it will call all the normal life cycle methods
  ✓ it warns when classic properties are defined on the instance, but does not invoke them.
  ✓ it should warn when misspelling shouldComponentUpdate
  ✓ it should warn when misspelling componentWillReceiveProps
  ✓ it should throw AND warn when trying to access classic APIs
  ✓ it supports this.context passed via getChildContext
  ✓ it supports classic refs
  ✓ it supports drilling through to the DOM using findDOMNode
```

whereas the old one made sure the number of assertions matched. So this is significantly less strict than before.